3rdpartylibrary/google-maps-android-cmgn/src/main/java/com/google/maps/android/kml/KmlContainerParser.java

```diff
@@ -47,22 +47,26 @@
     /**
      * Obtains a Container object (created if a Document or Folder start tag is read by the
      * XmlPullParser) and assigns specific elements read from the XmlPullParser to the container.
+     * @param documentBase Path leading to kml file directory to generate full path file url for map service.
+     *                     For example if the file was in C:/Documents/MyKMZ.kmz/test.kml this value would be C:/Documents/MyKMZ.kmz
      */
 
     /* package */
-    static KmlContainer createContainer(XmlPullParser parser)
+    static KmlContainer createContainer(XmlPullParser parser, String documentBase)
             throws XmlPullParserException, IOException {
-        return assignPropertiesToContainer(parser);
+        return assignPropertiesToContainer(parser, documentBase);
     }
 
     /**
      * Creates a new KmlContainer objects and assigns specific elements read from the XmlPullParser
      * to the new KmlContainer.
      *
      * @param parser XmlPullParser object reading from a KML file
+     * @param documentBase Path leading to kml file directory to generate full path file url for map service.
+     *                     For example if the file was in C:/Documents/MyKMZ.kmz/test.kml this value would be C:/Documents/MyKMZ.kmz
      * @return KmlContainer object with properties read from the XmlPullParser
      */
-    private static KmlContainer assignPropertiesToContainer(XmlPullParser parser)
+    private static KmlContainer assignPropertiesToContainer(XmlPullParser parser, String documentBase)
```

Comment:
I forgot to mention this before but please add the word final before all variables (for this part it should be in front of the parameters)